core/field_image.js

```diff
@@ -40,60 +40,60 @@ goog.require('Blockly.utils.Size');
  * @param {!(string|number)} width Width of the image.
  * @param {!(string|number)} height Height of the image.
  * @param {string=} opt_alt Optional alt text for when block is collapsed.
- * @param {Function=} opt_onClick Optional function to be called when the image
- *     is clicked.  If opt_onClick is defined, opt_alt must also be defined.
+ * @param {function(!Blockly.FieldImage)=} opt_onClick Optional function to be
+ *     called when the image is clicked. If opt_onClick is defined, opt_alt must
+ *     also be defined.
  * @param {boolean=} opt_flipRtl Whether to flip the icon in RTL.
+ * @param {Object=} opt_config A map of options used to configure the field.
+ *    See the [field creation documentation]{@link https://developers.google.com/blockly/guides/create-custom-blocks/fields/built-in-fields/image#creation}
+ *    for a list of properties this parameter supports.
  * @extends {Blockly.Field}
  * @constructor
  */
 Blockly.FieldImage = function(src, width, height,
-    opt_alt, opt_onClick, opt_flipRtl) {
-
+    opt_alt, opt_onClick, opt_flipRtl, opt_config) {
+  /* Return early. */
   if (!src) {
     throw Error('Src value of an image field is required');
   }
-
   if (isNaN(height) || isNaN(width)) {
     throw Error('Height and width values of an image field must cast to' +
       ' numbers.');
   }
-
-  // Ensure height and width are numbers.  Strings are bad at math.
   var imageHeight = Number(height);
   var imageWidth = Number(width);
   if (imageHeight <= 0 || imageWidth <= 0) {
     throw Error('Height and width values of an image field must be greater' +
       ' than 0.');
   }
 
-  /**
-   * Store the image height, since it is different from the field height.
-   * @type {number}
-   * @private
-   */
-  this.imageHeight_ = imageHeight;
-
+  /* Initialize configurable properties. */
   /**
    * Whether to flip this image in RTL.
    * @type {boolean}
    * @private
    */
-  this.flipRtl_ = opt_flipRtl || false;
+  this.flipRtl_ = false;
 
   /**
    * Alt text of this image.
    * @type {string}
    * @private
    */
-  this.altText_ = opt_alt || '';
-
-  if (typeof opt_onClick == 'function') {
-    this.clickHandler_ = opt_onClick;
-  }
+  this.altText_ = '';
 
   Blockly.FieldImage.superClass_.constructor.call(
-      this, src || '', null);
+      this, src || '', null, opt_config);
 
+  /* Give constructor parameters priority. */
+  if (opt_flipRtl != null) {
```

Comment:
Simply, 
```
if (opt_flipRtl)
```
as if it's false we don't need to re-set it.

Reply:
I was thinking if you had something like:

```
var defaultOptions = {
  alt: 'someAlt',
  flipRtl: true
}

// Override flipRtl to false.
var field  = new Blockly.FieldImage('src', 10, 10, null, null, false, defaultOptions);
```

You'd want the field's flipRtl value to be false. Similar thing with the alt field. (Admittedly, that's a pretty silly thing for people to do though.)

I'm cool with having the above result in true, or false, I just want confirmation first.

If we decide to have it result in true, I'll switch it and add tests to document (^-^)b

Reply:
Ah I see. Here are some thoughts on how the two ways of configuring interface together. 
- Currently, in some cases where the config param is configuring something that used to be configured via a constructor parameter, it's getting configured twice (as we're doing the old way first, and then overriding it). 
- In the interest of avoiding that, and avoiding developers having to guess which will win, we should only support one way or the other. ie: if the opt_config property doesn't exist, go ahead and use the constructor parameters and configuring properties in the constructor "the old way". Otherwise, ``configure_`` will take care of configuring parameters. 

What are your thoughts on that? Do you see reasons for supporting both simultaneously?

Reply:
Sounds like a great plan! I'll update everything accordingly :D 

Just gotta make sure it gets added to the docs.